staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go

```diff
@@ -264,6 +267,33 @@ func (c completedConfig) NewWithDelegate(delegationTarget genericapiserver.Deleg
 		return nil
 	})
 
+	if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.StorageVersionAPI) {
```

Comment:
What if the identity gate is not enabled?

Reply:
API servers will still update their storage versions and serve traffic. Stale storage version records won't get garbage collected. The storage migrator won't trigger migration unless those stale records get manually removed.

Reply:
Hm, sounds like we at least need to document that you shouldn't run like that.

Reply:
I added a commit to make enabling apiserver-identity a hard requirement to run this feature.